examples/mongocxx/get_values_from_documents.cpp

```diff
@@ -0,0 +1,127 @@
+//       {
+//         "uid" : 413098706,
+//         "status" : 2,
+//         "msg": "..."
+//       },
+//       ...
+//     ]
+// }
+
+// Construct a document in the format of 'messagelist'
+document::value new_message(int64_t uid, int32_t status, std::string msg) {
+    builder::stream::document doc;
+    return doc << "uid" << uid << "status" << status << "msg" << msg << builder::stream::finalize;
+}
+
+// Insert a document into the database
+void insert_test_data(mongocxx::collection& coll) {
+    builder::stream::document builder{};
+    builder << "messagelist" << builder::stream::open_array
+            << new_message(413098706, 3, "Lorem ipsum...")
+            << new_message(413098707, 2, "Lorem ipsum...")
+            << new_message(413098708, 1, "Lorem ipsum...") << builder::stream::close_array;
+
+    document::value doc = builder << builder::stream::finalize;
+
+    // Normally, one should check the return value for success
+    coll.insert_one(doc.view());
+}
+
+// Iterate over contents of messagelist
+void iterate_messagelist(document::element& ele) {
+    // Check validity && type before trying to iterate
+    if (ele.type() == type::k_array) {
+        array::view subarray{ele.get_array().value};
+        for (array::element msg : subarray) {
+            // Check correct type before trying to access elements.
+            // Only print out fields if they exist; don't report missing fields.
+            if (msg.type() == type::k_document) {
+                document::view subdoc{msg.get_document().value};
+                if (subdoc["uid"] && subdoc["uid"].type() == type::k_int64) {
+                    std::cout << "uid: " << subdoc["uid"].get_int64().value << std::endl;
+                }
+                if (subdoc["status"] && subdoc["status"].type() == type::k_int32) {
+                    std::cout << "status: " << subdoc["status"].get_int32().value << std::endl;
+                }
+                if (subdoc["msg"] && subdoc["msg"].type() == type::k_utf8) {
+                    std::cout << "msg: " << subdoc["msg"].get_utf8().value << std::endl;
+                }
+            } else {
+                std::cout << "Message is not a document" << std::endl;
+            }
+        }
+    } else {
+        std::cout << "messagelist is not an array" << std::endl;
+    }
+}
+
+// Print document parts to STDOUT
+void print_document(const bsoncxx::document::view& doc) {
+    // Define an element view for inspecting different fields
+    document::element ele{};
+
+    // Extract _id element as a string
+    ele = doc["_id"];
+    if (ele.type() == type::k_oid) {
+        std::string oid = ele.get_oid().value.to_string();
+        std::cout << "OID: " << oid << std::endl;
+    } else {
+        std::cout << "Error: _id was not an object ID." << std::endl;
+    }
+
+    // Extract "messagelist" element, which could be the 'invalid' (false)
+    // element if it doesn't exist
+    ele = doc["messagelist"];
```

Comment:
I find it slightly more stylish / easier to read to declare a new document::element variable here with a descriptive variable name, instead of reusing the one declared above at 88.  Suggest something like: deleting 88, replacing 91 with `document::element idEle = doc["_id"];`, and replacing 101 with `document::element messageListEle = doc["messageList"];`.